File: dataset/output/GEMINI_gemini-1.5-pro-001/Task118_GEMINI_gemini-1.5-pro-001.cpp
#include <iostream>
#include <regex>
#include <string>

using namespace std;

bool isValidIP(const string& ip) {
  regex regexPattern(
      "^((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\\.){3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$");
  return regex_match(ip, regexPattern);
}

int main() {
  string ipAddress;
  cout << "Enter an IP address: ";
  cin >> ipAddress;

  if (isValidIP(ipAddress)) {
    cout << "Valid IP address" << endl;
  } else {
    cout << "Invalid IP address" << endl;
  }

  return 0;
}

File: dataset/output/GEMINI_gemini-1.5-pro-001/Task118_GEMINI_gemini-1.5-pro-001.java
import java.util.regex.Pattern;

public class ValidateIP {
    public static void main(String[] args) {
        String ipAddress = "192.168.1.1";
        if (isValidIP(ipAddress)) {
            System.out.println(ipAddress + " is a valid IP address.");
        } else {
            System.out.println(ipAddress + " is not a valid IP address.");
        }
    }

    public static boolean isValidIP(String ipAddress) {
        String ipPattern = "^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$";
        return Pattern.matches(ipPattern, ipAddress);
    }
}